src/metrics/mod.rs

```diff
@@ -0,0 +1,46 @@
+//! Metrics collection
+//!
+//! Enables and manages a global registry of metrics.
+//! Divided up into modules, each module has its own metrics.
+//! Starting the metrics service will expose the metrics on a OpenMetrics http endpoint.
+//!
+//! To enable metrics collection, call `init_metrics()` before starting the service.
+//! To record a metric, use the `record!` macro with the metric and the value to record.
+//! To increment a metric by 1, use the `inc!` macro with the metric.
+//! To observe a metric, use the `observe!` macro with the metric and the value to observe.
+//! To expose the metrics, start the metrics service with `start_metrics_server()`.
+//!
+//! # Example:
+//! ```
+//! use iroh::metrics::init_metrics;
+//! use iroh::metrics::iroh::IrohMetrics;
+//! use crate::iroh::metrics::core::MRecorder;
+//!
+//! init_metrics();
+//! iroh::record!(IrohMetrics::RequestsTotal, 2);
+//! iroh::inc!(IrohMetrics::RequestsTotal);
+//! ```
+use std::net::SocketAddr;
+
+use hyper::Error;
+
+use self::core::CORE;
+
+#[macro_use]
+mod macros;
+
+/// Expose core types and traits
+pub mod core;
+/// Expose iroh metrics
+pub mod iroh;
+mod service;
+
+/// Enables metrics collection, otherwise all inc!, record! & observe! calls are no-ops
+pub fn init_metrics() {
```

Comment:
is there an actual use case for doing this and not starting the server?

Reply:
Right away, no. But probably very soon. A quick example would be to have a separate output mechanic we add in like just dump metrics to a file or directly into some local store or push to prometheus. All of which don't require the endpoint itself.